### PR TITLE
[4.0]fix error of twice clicking on toolbar and its disappearance on resizing

### PR DESCRIPTION
**Pull request by Subhang23**
Pull Request for Issue #26543 and #27957 .

### Summary of Changes
Added class to the div so that on loading it stays in either collapsed form or not-collapsed form(previously it was in a state of mix).
Added querySelectors to ensure that whenever screen size is that of desktop the classes of toolbar stays the same.

### Testing Instructions
Go to any page which has a toolbar and reduce the width of the screen to a mobile size.

### Expected result
You can open or close the toolbar with a single click and on resizing and closing the toolbar and again resizing the toolbar is intact.

### Actual result
With the initial state as pulled up only one click is required. Even on removing toolbar in mobile-view and resizing the toolbar is seen

### CODE REVIEW

**Comment by Subhang23** [comment]
Screenshot from 2020-02-17 11-39-17

**Comment by SharkyKZ** [comment]
Now the toolbar is missing on large screens.

**Comment by Subhang23** [comment]
I understood the error it was because if the toolbar is compacted in mobile-view it is hidden in desktop-view. This might also be the reason for Issue #27957 .

**Comment by infograf768** [comment]
Looks like working.
Now the toolbar is showing on desktop
Note: If the screen is reduced, the toolbar closed and then the screen enlarged to desktop, it requires a reload to get the toolbar again.
Issue or not?

**Comment by SharkyKZ** [comment]
That's the issue reported in #27957.

**Comment by Subhang23** [comment]
> That's the issue reported in #27957.

I am currently working on it.

**Comment by Subhang23** [comment]
> Looks like working.
> Now the toolbar is showing on desktop
> Note: If the screen is reduced, the toolbar closed and then the screen enlarged to desktop, it requires a reload to get the toolbar again.
> Issue or not?

Is this fine?

**Comment by infograf768** [comment]
> Is this fine?

When https://github.com/joomla/joomla-cms/issues/27957 is solved, yes.

**Comment by Subhang23** [comment]
So should I close this PR and open a PR that solves this issue and Issue #27957 together. I thought they can be solved separately, so I thought of opening 2 separate PR's.
What should I do @infograf768 ?

**Comment by infograf768** [comment]
I am not the one to decide.
If you can solve it together here or in a new PR, it is easier on testers.

**Comment by Subhang23** [comment]
@infograf768 I have fixed both the issues with this PR. Please review it.

**Comment by brianteeman** [comment]
Looks good to me

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on a02351f16fbb7bfb80ac4c06cfd74f81db0b104b<br /><br />Great!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27958">issues.joomla.org/tracker/joomla-cms/27958</a>.</sub>

File: build/media_source/templates/atum/js/template.es6.js [hunk 1]
```diff
@@ -237,6 +237,12 @@
    */
   function reactToResize() {
     window.addEventListener('resize', () => {
+      if (window.innerWidth >= 576) {
+        const toolbar = doc.querySelector('.container-fluid');
+        toolbar.getElementsByClassName('subhead')[0].classList.add('show');
```
Comment (C-Lodder) [inline build/media_source/templates/atum/js/template.es6.js]: `toolbar.querySelector('#subhead').classList.add('show');`

File: build/media_source/templates/atum/js/template.es6.js [hunk 2]
```diff
@@ -237,6 +237,12 @@
    */
   function reactToResize() {
     window.addEventListener('resize', () => {
+      if (window.innerWidth >= 576) {
+        const toolbar = doc.querySelector('.container-fluid');
```
Comment (C-Lodder) [inline build/media_source/templates/atum/js/template.es6.js]: Target `.container-main`, not `.container-fluid`. Or better yet, add an ID to the `<div>` and target that with `document.getElementById('xxxx')`

File: build/media_source/templates/atum/js/template.es6.js [hunk 3]
```diff
@@ -237,6 +237,12 @@
...
+        toolbar.getElementsByClassName('subhead')[0].classList.add('show');
+        toolbar.getElementsByClassName('navbar-toggler toggler-toolbar')[0].setAttribute('aria-expanded', 'true');
```
Comment (C-Lodder) [inline build/media_source/templates/atum/js/template.es6.js]: `toolbar.querySelector('.toggler-toolbar').setAttribute('aria-expanded', true);`

File: build/media_source/templates/atum/js/template.es6.js [hunk 4]
```diff
@@ -237,6 +237,12 @@
...
+        toolbar.getElementsByClassName('navbar-toggler toggler-toolbar')[0].classList.remove('collapsed');
```
Comment (C-Lodder) [inline build/media_source/templates/atum/js/template.es6.js]: `toolbar.querySelector('.toggler-toolbar').classList.remove('collapsed');`

**Comment by Subhang23** [comment]
@C-Lodder I have made the required changes.

File: administrator/templates/atum/index.php [hunk 5]
```diff
@@ -137,7 +137,7 @@
 	<?php endif; ?>
 
 	<?php // container-fluid ?>
-	<div class="container-fluid container-main">
+	<div class="container-fluid container-main" id="main">
```
Comment (infograf768) [inline administrator/templates/atum/index.php]: Not sure using "main" as id is a good idea.

Comment (Subhang23) [reply administrator/templates/atum/index.php]: @infograf768 Would it be fine if I used main-container as the id name

Comment (infograf768) [reply administrator/templates/atum/index.php]: better indeed.

Comment (Subhang23) [reply administrator/templates/atum/index.php]: @infograf768 Done. Please check it.

**Comment by C-Lodder** [comment]
So right now, every time you change the viewport size by 1px, you're going to be searching to DOM 5 times. Not very good for performance.

**Comment by C-Lodder** [comment]
@Subhang23 targetting DOM elements should go outside of the `resize` event listener.

File: administrator/templates/atum/index.php [hunk 6]
```diff
@@ -137,13 +137,13 @@
 	<?php endif; ?>
```
Comment (wilsonge) [inline administrator/templates/atum/index.php:137]: Can you fix the permissions on this file please?

Comment (Subhang23) [reply administrator/templates/atum/index.php:137]: I will get this done

**Comment by Subhang23** [comment]
@wilsonge please review it

File: build/media_source/templates/atum/js/template.es6.js [hunk 7]
```diff
@@ -236,7 +236,15 @@
    * @since   4.0.0
    */
   function reactToResize() {
+    const toolbar = doc.getElementById('main-container');
     window.addEventListener('resize', () => {
+      if (window.innerWidth >= 576) {
+        if (toolbar && !toolbar.querySelector('#subhead').classList.contains('show')) {
```
Comment (wilsonge) [inline build/media_source/templates/atum/js/template.es6.js:242]: In cpanel `#subhead` doesn't exist and this causes a JS error getting the property classList of null

**Comment by wilsonge** [comment]
This doesn't work for me. First of all it totally fails on cpanel (see comment above).

When the menu is open and i leave mobile menu it stays open. However when the mobile menu is hidden it doesn't reappear at all. See screenshot - there's no way to reopen the menu at all
Screenshot 2020-04-04 at 17 49 57

**Comment by alikon** [comment]
same from my test

**Comment by Subhang23** [comment]
I am not sure if this is the right place to ask but I have run into a problem while setting up Joomla on my local server
Screenshot from 2020-04-05 01-50-32
I am running it on Ubuntu 19.04 php7.3

**Comment by wilsonge** [comment]
not sure what's going on there :( looks very sad. are you sure Joomla is even loading and it's not something from the web server. all our messages should have more info than that

**Comment by particthistle** [comment]
I have tested this item :red_circle: unsuccessfully on 32c00cbcac249f76047b356f9c57ec36386f7b6f<br /><br />Patch was unable to be applied. Receiving:

Error
The patch could not be applied because the repository is missing

_@Subhang23 if you're still having issues with your local installation, read https://magazine.joomla.org/all-issues/june-2020/github-installing-git which might reveal something you've done wrong with your local installation, though this article outlines XAMPP not Ubuntu specifically._<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27958">issues.joomla.org/tracker/joomla-cms/27958</a>.</sub>

**Comment by Quy** [comment]
Please test PR #30131

**Pull request closed without merging**